server/src/features/facilities/facility.service.js

```diff
@@ -1,70 +1,123 @@
 import { CourtBooking, GymSession } from "./facility.model.js";
-import { sendGymSessionCancellationEmail, sendGymSessionUpdateEmail } from "../auth/email.service.js";
+import {
+  sendGymSessionCancellationEmail,
+  sendGymSessionUpdateEmail,
+} from "../auth/email.service.js";
 
 class FacilitiesServiceClass {
-async getCourtSchedules() {
-  const today = new Date();
-  today.setHours(0, 0, 0, 0);
+  async reserveCourt(userId, reservationData) {
+    const { courtType, date, startTime, endTime } = reservationData;
```

Comment:
```suggestion
     const user = await User.findById(userId);
     if (!user) throw new Error("Invalid user");

    const { courtType, date, startTime, endTime } = reservationData;
```
check that the user exists if the a deleted user somehow sends a token they can still book